lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -443,146 +517,243 @@ def create_autoscaling_group(connection, module):
     health_check_type = module.params.get('health_check_type')
     default_cooldown = module.params.get('default_cooldown')
     wait_for_instances = module.params.get('wait_for_instances')
-    as_groups = connection.get_all_groups(names=[group_name])
+    as_groups = connection.describe_auto_scaling_groups(AutoScalingGroupNames=[group_name])
```

Comment:
Strictly, they should also be using AWSRetry as well, but that can be left to a followup PR.